packages/uploadthing/src/client.ts

```diff
@@ -122,6 +130,28 @@ export const genUploader = <TRouter extends FileRouter>(
     );
 };
 
+interface Done {
+  status: "done";
+  callbackData: unknown;
+}
+interface StillWaiting {
+  status: "still waiting";
+}
+type PollingResponse = Done | StillWaiting;
+
+const isPollingResponse = (input: unknown): input is PollingResponse =>
+  input != null && // @datner: != is not accidental
+  typeof input === "object" &&
+  "status" in input &&
```

Comment:
we have an `isObject` util which should allow you to do `isObject(input) && typeof input.status === "string"`